simpleble/src/backends/linux/AdapterLinux.cpp

```diff
@@ -1,46 +1,29 @@
+#include <thread>
+
+#include <simpleble/Peripheral.h>
+
 #include "AdapterLinux.h"
-#include "BackendBluez.h"
 #include "BuildVec.h"
 #include "BuilderBase.h"
 #include "CommonUtils.h"
 #include "PeripheralLinux.h"
 
 using namespace SimpleBLE;
 
-std::vector<std::shared_ptr<AdapterBase>> AdapterBase::get_adapters() {
-    std::vector<std::shared_ptr<AdapterBase>> adapter_list;
-    auto internal_adapters = Bluez::get()->bluez.get_adapters();
-    for (auto& adapter : internal_adapters) {
-        adapter_list.push_back(std::make_shared<AdapterBase>(adapter));
-    }
-    return adapter_list;
-}
-
-bool AdapterBase::bluetooth_enabled() {
-    bool enabled = false;
-
-    auto internal_adapters = Bluez::get()->bluez.get_adapters();
-    for (auto& adapter : internal_adapters) {
-        if (adapter->powered()) {
-            enabled = true;
-            break;
-        }
-    }
-
-    return enabled;
-}
+bool AdapterLinux::bluetooth_enabled() { return adapter_->powered(); }
 
-AdapterBase::AdapterBase(std::shared_ptr<SimpleBluez::Adapter> adapter) : adapter_(adapter) {}
+AdapterLinux::AdapterLinux(std::shared_ptr<SimpleBluez::Adapter> adapter) : adapter_(adapter) {}
 
-AdapterBase::~AdapterBase() { adapter_->clear_on_device_updated(); }
+AdapterLinux::~AdapterLinux() { adapter_->clear_on_device_updated(); }
 
-void* AdapterBase::underlying() const { return adapter_.get(); }
+// TODO: why is this necessary?
+void* AdapterLinux::underlying() const { return adapter_.get(); }
```

Comment:
You can add a developer note for this one. For certain compatibility with external libraries, we sometimes need to expose the actual OS handle to the user. This is particularly important for MacOS right now, but we might do something different in the future.

Reply:
Done: https://github.com/OpenBluetoothToolbox/SimpleBLE/pull/370/files#diff-c71b60d64f24eb87b27f2a6647d187d46e4f5b86f7ea1680f70d7c706002d133R34